### PR TITLE
Added interested levels support

### PR DESCRIPTION
**Pull request by ilyaplot**
Earlier definition of the list of levels of logging for target in any way did not influence its work.

### CODE REVIEW

**Comment by samdark** [comment]
It was passing messages to PSR logger as is.

**Comment by ilyaplot** [comment]
> It was passing messages to PSR logger as is.

The list of levels can be specified in the config, but this does not lead to anything. I thought that it's better to implement the functionality than to mislead the developer. I tried to use SlackHandler only for certain levels, but I could not. Now this problem is solved.

**Comment by ilyaplot** [comment]
You can, of course, define levels in the Rsr handler, but if I want to log only the TRACE level, then I get PROFILE levels too.

**Comment by samdark** [comment]
Yes. It makes sense to add it. Please fix code (see travis build fail).

File: src/PsrTarget.php [hunk 1]
```diff
@@ -39,6 +39,22 @@ class PsrTarget extends Target implements LoggerAwareInterface
         LogLevel::DEBUG => LogLevel::DEBUG,
     ];
 
+    /**
+     * @var array
+     */
+    private $_interestedLevels = null;
```
Comment (samdark) [inline src/PsrTarget.php]: No need to specify `= null`.

File: src/PsrTarget.php [hunk 2]
```diff
@@ -39,6 +39,22 @@ class PsrTarget extends Target implements LoggerAwareInterface
...
+
+    /**
+     * @inheritdoc
+     */
+    public function init()
+    {
+        parent::init();
+        if (is_null($this->_interestedLevels)) {
```
Comment (samdark) [inline src/PsrTarget.php]: No need for this check. At `init()` state it will be null all the time.

File: src/PsrTarget.php (hunk 1 above)
Comment (samdark) [reply src/PsrTarget.php]: Could be named `_levels` similar to base `Target` class.

File: src/PsrTarget.php [hunk 3]
```diff
@@ -39,6 +39,22 @@ class PsrTarget extends Target implements LoggerAwareInterface
...
+            $this->_interestedLevels = $this->_psrLevels;
```
Comment (samdark) [inline src/PsrTarget.php]: I think that by default there should be both PSR and non-PSR levels. Would simplify code significantly.

Comment (ilyaplot) [reply src/PsrTarget.php]: I did not quite understand the point. You mean the division into PSR and non-PSR levels?

Comment (samdark) [reply src/PsrTarget.php]: I mean `$this->_interestedLevels = [all PSR and Yii levels here]`

Comment (ilyaplot) [reply src/PsrTarget.php]: In this case, _psrLevels will duplicate _levels. I left only _levels.

File: src/PsrTarget.php [hunk 4]
```diff
@@ -81,7 +88,61 @@ public function export()
...
+     * ```php
+     * ['error', 'warning', LogLevel::CRITICAL, LogLevel::EMERGENCY]
+     * ```
+     *
+     * @param array $levels message levels that this target is interested in.
+     * @throws InvalidConfigException if $levels value is not correct.
+     */
+    public function setLevels($levels)
+    {
+        static $levelMap = [
```
Comment (samdark) [inline src/PsrTarget.php:118]: That would disallow using PSR levels in setLevels(). Is that intended?

Comment (ilyaplot) [reply src/PsrTarget.php:118]: Why? _levels contains the PSR levels. I need to write tests for this.

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**